File: backend/app/schemas/scheduling.py
"""Scheduled-order (collection slot) Pydantic schemas."""

from datetime import datetime

from pydantic import BaseModel


class PickupSlot(BaseModel):
    at: datetime
    remaining: int | None  # None = no per-slot cap set


class PickupSlotsResponse(BaseModel):
    date: str          # the local date these slots belong to
    timezone: str      # the restaurant's own zone, so a client can label the times correctly
    days_ahead: int    # how far ahead this restaurant takes pre-orders
    interval_minutes: int
    slots: list[PickupSlot]

File: frontend/app/types/scheduling.ts
export type PickupSlot = {
  at: string
  /** null = no per-slot cap set by the restaurant. */
  remaining: number | null
}

export type PickupSlots = {
  date: string
  /** The restaurant's own zone — label the times in it, not the visitor's. */
  timezone: string
  days_ahead: number
  interval_minutes: number
  slots: PickupSlot[]
}
